### PR TITLE
[Merged by Bors] - feat(MeasureTheory): generalize `ae` to `OuterMeasureClass`

### PR DESCRIPTION
**Pull request by urkud**
- Move definition and some lemmas from `MeasureSpaceDef` to `OuterMeasure.AE`.
- Rename `MeasureTheory.Measure.ae` to `MeasureTheory.ae`.
- Generalize the definition and theorems to `[OuterMeasureClass _ _]`.

See also https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/.22Almost.20every.22.20on.20outer.20measures and https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Order.20on.20measures/near/425729507

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by j-loreaux** [comment]
@urkud wasn't there a Zulip thread about this? Can you link to that please in the PR description?

**Comment by urkud** [comment]
I missed that discussion, found and added a link.

File: Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean [hunk 1]
```diff
@@ -626,14 +378,14 @@ section MeasureSpace
 
 This is notation for `Filter.Eventually P (Measure.ae MeasureSpace.volume)`. -/
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:379]:
```suggestion
This is notation for `Filter.Eventually P (MeasureTheory.ae MeasureSpace.volume)`. -/
```

File: Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean [hunk 2]
```diff
@@ -626,14 +378,14 @@ section MeasureSpace
...
 notation3 "∀ᵐ "(...)", "r:(scoped P =>
-  Filter.Eventually P <| MeasureTheory.Measure.ae MeasureTheory.MeasureSpace.volume) => r
+  Filter.Eventually P <| MeasureTheory.ae MeasureTheory.MeasureSpace.volume) => r
 
 /-- `∃ᵐ a, p a` means that `p` holds frequently, i.e. on a set of positive measure,
 w.r.t. the volume measure.
 
 This is notation for `Filter.Frequently P (Measure.ae MeasureSpace.volume)`. -/
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:386]:
```suggestion
This is notation for `Filter.Frequently P (MeasureTheory.ae MeasureSpace.volume)`. -/
```

File: Mathlib/MeasureTheory/OuterMeasure/AE.lean [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+If `μ` is an outer measure or a measure on `α`,
+then `MeasureTheory.ae μ` is the filter of co-null sets: `s ∈ ae μ ↔ μ sᶜ = 0`.
+
+In this file we define the filter and prove some basic theorems about it.
+
+## Notation
+
+- `∀ᵐ x ∂μ, p x`: predicate `p` holds for `μ`-a.e. all `x`;
+- `∃ᶠ x ∂μ, p x`: predicate `p` holds on a set of nonzero measure;
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/OuterMeasure/AE.lean:19]:
```suggestion
- `∀ᵐ x ∂μ, p x`: the predicate `p` holds for `μ`-a.e. all `x`;
- `∃ᶠ x ∂μ, p x`: the predicate `p` holds on a set of nonzero measure;
```

**Review by sgouezel — COMMENTED** [review]
bors d+
That's great, thanks!

**Pull request closed without merging**